src/script/mod.rs

```diff
@@ -530,34 +535,40 @@ impl Output {
 }
 
 /// An AsyncRead wrapper that replaces carriage return (\r) bytes with newline (\n) bytes.
-struct CarriageReturnToNewline<R: AsyncRead + Unpin> {
-    inner: R,
+pub fn normalize_crlf<R: AsyncRead + Unpin>(reader: R) -> impl AsyncRead + Unpin {
+    FramedRead::new(reader, CrLfNormalizer::default())
+        .into_async_read()
+        .compat()
 }
 
-impl<R: AsyncRead + Unpin> AsyncRead for CarriageReturnToNewline<R> {
-    fn poll_read(
-        mut self: Pin<&mut Self>,
-        cx: &mut Context<'_>,
-        buf: &mut ReadBuf<'_>,
-    ) -> Poll<std::io::Result<()>> {
-        let initial_filled = buf.filled().len();
-
-        // Poll the inner reader
-        match Pin::new(&mut self.inner).poll_read(cx, buf) {
-            Poll::Ready(Ok(())) => {
-                // Get a mutable slice of the *newly* filled part of the buffer
-                let newly_filled = &mut buf.filled_mut()[initial_filled..];
-
-                for byte in newly_filled.iter_mut() {
-                    if *byte == b'\r' {
-                        *byte = b'\n';
-                    }
+/// Codec that normalizes CR and CRLF to LF
+#[derive(Default)]
+struct CrLfNormalizer {
+    last_was_cr: bool,
+}
+
+impl Decoder for CrLfNormalizer {
+    type Item = BytesMut;
+    type Error = io::Error;
+
+    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        if src.is_empty() {
+            return Ok(None);
+        }
+        let mut bytes = src.split_off(0);
+        let mut w = 0;
+        for r in 0..bytes.len() {
+            self.last_was_cr = match (bytes[r], self.last_was_cr) {
+                (b'\n', true) => false,
+                (b, _) => {
```

Comment:
You are again mixing match and if statements here. I find that harder to understand. Just use multiple matches. You are also checking if the byte is a CR twice.